src/AWS.Deploy.Common/Recipes/Validation/RegexValidator.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Text.RegularExpressions;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    public class RegexValidator : IOptionSettingItemValidator
+    {
+        public string Regex { get; set; } = "(.*)";
+        public string ValidationFailedMessage { get; set; } = "Value must match Regex {{Regex}}";
+        public bool AllowEmptyString { get; set; }
+
+        public ValidationResult Validate(object input)
+        {
+            var regex = new Regex(Regex);
+
+            var message = ValidationFailedMessage.Replace("{{Regex}}", Regex);
+
+            return new ValidationResult
+            {
+                IsValid = regex.IsMatch(input?.ToString() ?? "") || (AllowEmptyString && string.IsNullOrEmpty(input?.ToString())),
```

Comment:
input?.ToString() can be null. Then IsMatch would throw an exception. Handle the null to avoid unexpected excetions.

Reply:
The null-coalescing operator(??) will return an empty string if `input?.ToString()` is null.